### ISSUE TITLE
Highlight thread_local as a keyword

### ISSUE THREAD

**Issue by jgcodes2020** [clangd/clangd#841]
C++'s `thread_local` keyword is used to designate a variable which has _thread storage duration_ (i.e. having one copy of the variable per thread, that is destructed when the thread terminates). 

Since it is a keyword, it should be highlighted like one.
```c++
void do_thing() {
  thread_local int thing;
}
```
GitHub's syntax highlighting already does this, and it'd be nice to have this in clangd.

**Comment by HighCommander4** [comment]
Keyword highlighting is handled on the client side, not by clangd.

(In vscode, I use [this plugin](https://github.com/jeff-hykin/cpp-textmate-grammar) for client-side C++ highlighting, and this bug report made me realize that it indeed misses `thread_local`, so I filed https://github.com/jeff-hykin/cpp-textmate-grammar/issues/561).

**Comment by kadircet** [comment]
Yes we deliberately don't highlight keywords (and probably won't). As editors' text-based syntax highlighting works fast and correct for those cases already and highlighting those through clangd will probably result in bad UX overall (flickering/glitches)
